app/src/androidTest/java/org/kiwix/kiwixmobile/nav/destination/reader/ReaderRobot.kt

```diff
@@ -15,16 +15,22 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-package org.kiwix.kiwixmobile.main
 
+package org.kiwix.kiwixmobile.nav.destination.reader
+
+import applyWithViewHierarchyPrinting
 import org.kiwix.kiwixmobile.BaseRobot
-import org.kiwix.kiwixmobile.Findable.ViewId
+import org.kiwix.kiwixmobile.Findable.StringId.TextId
 import org.kiwix.kiwixmobile.R
 
-fun main(func: MainRobot.() -> Unit) = MainRobot().apply(func)
+/**
+ * Authored by Ayush Shrivastava on 29/10/20
+ */
+
+fun reader(func: ReaderRobot.() -> Unit) = ReaderRobot().applyWithViewHierarchyPrinting(func)
 
-class MainRobot : BaseRobot() {
+class ReaderRobot : BaseRobot() {
   init {
-    isVisible(ViewId(R.id.navigation_container))
+    isVisible(TextId(R.string.reader))
```

Comment:
not a valid proof

Reply:
I'm trying to use assertion on one of the options displayed in the main menu(the one which contains search, full screen, reading aloud, tab switcher, rand. article etc.), but the thing is that when we are inflating the menu layout in the reader screen, we make only a few options of it visble, rest are invisible. Is this why robot fails clicking the `R.menu.menu_main` or I'm inferring something wrong at some point?

Reply:
Why is this relevant here? Is the robot failing to click the options menu? Why are we clicking it?
If you want to assert the Reader screen is visible there are many properties to choose from such as `R.id.activity_main_root`. Just look to the xml that is being inflated

Reply:
> Why is this relevant here?

Ah, using the layout for assertion didn't click me that's why I was trying to assert the views, sorry for that.
>  Is the robot failing to click the options menu? 

I use clickOn and then pass the menu in ViewId param of clickOn, on doing this, it is unable to find the menu to click.

Reply:
The options menu doesn't really have an id we can rely on, that is why we use ContentDescriptions aka `ContentDesc` when we want to click on them elsewhere. Fore fine grained information I suggest using the [UiAutomator Viewer](https://developer.android.com/training/testing/ui-automator#ui-automator-viewer)